File: 2018/2/part2.ts
const readline = require('readline');
const fs = require('fs');

const rl = readline.createInterface({
  input: fs.createReadStream('input'),
  crlfDelay: Infinity
});

var lines: Array<Array<number>> = [];
rl.on('line', (line) => {
  let linearray = line.split('');
  lines.push(linearray)
});

rl.on('close', () => {
    for (var i of lines){
        for (var j of lines){
            var diffs: number = 0;
            var matchingIndex: number = 0
            i.map((char, index)=>{
                if (char != j[index]){
                    diffs++;
                    matchingIndex = index;
                }
            })
            if (diffs === 1){
                i.splice(matchingIndex,1);
                console.log(i.join(''));
                process.exit(0)
            }
        }
    }
});

File: 2020/1/part2.py
import sys

with open("input",'r') as file:
    lines = [int(x) for x in file.read().splitlines()]

for line in lines:
    for line2 in lines:
        for line3 in lines:
            if line+line2+line3 == 2020:
                print(line*line2*line3)
                sys.exit(0)